cgi-bin/LJ/Talk.pm

```diff
@@ -2831,6 +2831,270 @@ sub prepare_and_validate_comment {
     return $comment;
 }
 
+# Handles user auth for the talkform's "from" fields.
+# Args:
+# - $form: a hashref representing the POSTed comment form. It might Go Through
+# Some Changes during this function (mostly affecting those "from" fields).
+# - $remote: the current logged-in LJ::User, or undef, OR the just-now authenticated OpenID
+# user (which is why we can't just get_remote like normal).
+# - $journalu: LJ::User who owns the journal the comment was submitted to. Need
+# this for storing pending comments in first pass through openid auth, and also
+# we use it to switch off between variant error messages.
+# - $errret: an array ref to push errors onto. Actually, should probably
+# refactor this to just return an error instead; we only ever push one error,
+# since any error here is fatal.
+# - $didlogin: an optional scalar ref to indicate whether we started a new login
+# session. Only really used for an informational thing on an interstitial page
+# after commenting, which doesn't appear 100% reliably.
+# Returns one of:
+# - user object (authenticated user)
+# - undef (anon)
+# - string with URL (openid first pass)
+sub authenticate_user_and_mutate_form {
```

Comment:
I appreciate the descriptive names.

Reply:
Thanks, I tried. (Partly I was just still mad about the vacuousness of "init", lol)